pkg/dockerregistry/server/prune/prune.go

```diff
@@ -101,10 +213,7 @@ func Prune(ctx context.Context, storageDriver driver.StorageDriver, registry dis
 		is, err := oc.ImageStreams(ref.Namespace).Get(ref.Name, metav1.GetOptions{})
 		if kerrors.IsNotFound(err) {
 			logger.Printf("The image stream %s/%s is not found, will remove the whole repository", ref.Namespace, ref.Name)
-
-			// We cannot delete the repository at this point, because it would break Enumerate.
-			reposToDelete = append(reposToDelete, repoName)
-
+			gc.SetRepository(repoName)
```

Comment:
Explain why the deletion needs to be delayed.

Reply:
yes please doc the whole logic of how this deferred deletion works (and why it is necessary).